bi/pkg/cluster/pulumi.go

```diff
@@ -23,54 +24,49 @@ const (
 
 type pulumiProvider struct {
 	initSuccessful bool
+	spec           *specs.InstallSpec
 
 	cfg auto.ConfigMap
 	projectName,
-	slug,
 	workDirRoot string
 
 	pulumiHome auto.LocalWorkspaceOption
 	pulumi     auto.LocalWorkspaceOption
 	envVars    auto.LocalWorkspaceOption
 }
 
-func NewPulumiProvider(slug string) Provider {
+func NewPulumiProvider(spec *specs.InstallSpec) Provider {
 	return &pulumiProvider{
 		projectName: "bi",
-		slug:        slug,
+		spec:        spec,
 	}
 }
 
 func (p *pulumiProvider) Init(ctx context.Context) error {
-	ws, err := p.configure(ctx)
+	_, err := p.configure(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to configure pulumi provider: %w", err)
 	}
 
-	// add plugins that need to be installed here
-	plugins := map[string]string{
-		"aws":       "v6.54.2",
-		"cloudinit": "v1.4.7",
-		"tls":       "v5.0.7",
-	}
-	if err := p.installPlugins(ctx, ws, plugins); err != nil {
-		return fmt.Errorf("failed to install necessary pulumi plugins: %w", err)
-	}
-
```

Comment:
These weren't kept in sync with the versions in `go.mod` so we were installing these versions and pulumi was installing the version matching `go.mod` anyways.